federation/pkg/federatedtypes/namespace.go

```diff
@@ -0,0 +1,215 @@
+}
+
+func (a *NamespaceAdapter) FedCreate(obj pkgruntime.Object) (pkgruntime.Object, error) {
+	namespace := obj.(*apiv1.Namespace)
+	return a.client.CoreV1().Namespaces().Create(namespace)
+}
+
+func (a *NamespaceAdapter) FedDelete(qualifiedName QualifiedName, options *metav1.DeleteOptions) error {
+	return a.client.CoreV1().Namespaces().Delete(qualifiedName.Name, options)
+}
+
+func (a *NamespaceAdapter) FedGet(qualifiedName QualifiedName) (pkgruntime.Object, error) {
+	return a.client.CoreV1().Namespaces().Get(qualifiedName.Name, metav1.GetOptions{})
+}
+
+func (a *NamespaceAdapter) FedList(namespace string, options metav1.ListOptions) (pkgruntime.Object, error) {
+	return a.client.CoreV1().Namespaces().List(options)
+}
+
+func (a *NamespaceAdapter) FedUpdate(obj pkgruntime.Object) (pkgruntime.Object, error) {
+	namespace := obj.(*apiv1.Namespace)
+	return a.client.CoreV1().Namespaces().Update(namespace)
+}
+
+func (a *NamespaceAdapter) FedWatch(namespace string, options metav1.ListOptions) (watch.Interface, error) {
+	return a.client.CoreV1().Namespaces().Watch(options)
+}
+
+func (a *NamespaceAdapter) ClusterCreate(client kubeclientset.Interface, obj pkgruntime.Object) (pkgruntime.Object, error) {
+	namespace := obj.(*apiv1.Namespace)
+	return client.CoreV1().Namespaces().Create(namespace)
+}
+
+func (a *NamespaceAdapter) ClusterDelete(client kubeclientset.Interface, qualifiedName QualifiedName, options *metav1.DeleteOptions) error {
+	return client.CoreV1().Namespaces().Delete(qualifiedName.Name, options)
+}
+
+func (a *NamespaceAdapter) ClusterGet(client kubeclientset.Interface, qualifiedName QualifiedName) (pkgruntime.Object, error) {
+	return client.CoreV1().Namespaces().Get(qualifiedName.Name, metav1.GetOptions{})
+}
+
+func (a *NamespaceAdapter) ClusterList(client kubeclientset.Interface, namespace string, options metav1.ListOptions) (pkgruntime.Object, error) {
+	return client.CoreV1().Namespaces().List(options)
+}
+
+func (a *NamespaceAdapter) ClusterUpdate(client kubeclientset.Interface, obj pkgruntime.Object) (pkgruntime.Object, error) {
+	namespace := obj.(*apiv1.Namespace)
+	return client.CoreV1().Namespaces().Update(namespace)
+}
+
+func (a *NamespaceAdapter) ClusterWatch(client kubeclientset.Interface, namespace string, options metav1.ListOptions) (watch.Interface, error) {
+	return client.CoreV1().Namespaces().Watch(options)
+}
+
+func (a *NamespaceAdapter) IsSchedulingAdapter() bool {
+	return false
+}
+
+func (a *NamespaceAdapter) NewTestObject(namespace string) pkgruntime.Object {
+	return &apiv1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-namespace-",
+		},
+		Spec: apiv1.NamespaceSpec{
+			Finalizers: []apiv1.FinalizerName{apiv1.FinalizerKubernetes},
+		},
+	}
+}
+
+// CleanupNamespace deletes all resources in a given namespace.
+func (a *NamespaceAdapter) CleanupNamespace(obj pkgruntime.Object, eventRecorder record.EventRecorder) (pkgruntime.Object, error) {
```

Comment:
Should there be a test for this? I suppose it's not that easy to test because of the interface and the collaborators.